pkg/apis/serving/k8s_validation.go

```diff
@@ -350,7 +350,17 @@ func validateInitContainers(ctx context.Context, containers []corev1.Container,
 		return errs.Also(&apis.FieldError{Message: fmt.Sprintf("pod spec support for init-containers is off, "+
 			"but found %d init containers", len(containers))})
 	}
+	allNames := sets.String{}
```

Comment:
nit: We can probably preallocate this, like `allNames := make(sets.String, len(otherContainers)+len(containers))`